read-fonts/src/tables/glyf.rs

```diff
@@ -55,34 +54,33 @@ impl<'a> SimpleGlyph<'a> {
     }
 }
 
+/// Point for a simple glyph.
 #[derive(Clone, Copy, Debug)]
 pub struct Point {
+    /// X component.
     pub x: i16,
+    /// Y component.
     pub y: i16,
+    /// True if this is an on-curve point.
+    pub on_curve: bool,
 }
 
-#[derive(Clone, Copy, Debug)]
-pub enum GlyphPoint {
-    OffCurve(Point),
-    OnCurve(Point),
-    End(Point),
-}
-
-pub struct PointIter<'a> {
+#[derive(Clone)]
+struct PointIter<'a> {
     end_points: &'a [BigEndian<u16>],
     cur_point: u16,
-    flags: OldCursor<'a>,
-    x_coords: OldCursor<'a>,
-    y_coords: OldCursor<'a>,
+    flags: Cursor<'a>,
+    x_coords: Cursor<'a>,
+    y_coords: Cursor<'a>,
```

Comment:
in case this wasn't obvious: this glyf code predates the current codegen code, and the current implementation of Cursor. I ported it over at some point but found it annoying to express with the existing cursor, and ended up just using the simple cursor I had implemented at the time for that purpose.

Reply:
The only real difference seems to be the Option/Result distinction. We may want to surface read errors in the glyph parser at some point. I just figured I'd get rid of `OldCursor` while I was in the code.